test/calltoaction.spec.js

```diff
@@ -1,151 +1,114 @@
-      });
+          it('is derived from the HTML element\'s id and data-cookie-name-root attribute', function () {
+            const id = fixtures.generateRandomId();
+            const $fixture = fixtures.generateHTMLWithCookieDetails(fixtures.getCookieNameRoot(), id);
+            const callToAction = new CallToAction($fixture);
+            callToAction.dismissible.$button.click();
+            const expectedCookieName = `${fixtures.getCookieNameRoot()}${id}`;
+            expect(callToAction.dismissible.cookieName).to.equal(expectedCookieName);
+            expect(utils.getCookieValue(expectedCookieName, document.cookie)).to.equal('true');
+            fixtures.clearCookie(id);
+          });
 
-    });
+          it('is empty if the HTML element lacks an id', function () {
+            const $fixture = fixtures.generateHTML();
+            $fixture.dataset.cookieNameRoot = fixtures.getCookieNameRoot();
 
-    describe('the cookie name', () => {
+            const callToAction = new CallToAction($fixture);
+            expect(callToAction.dismissible.cookieName).to.equal('');
 
-      it('is the string "callToAction_" appended with the id attribute of the component\'s HTML element', () => {
-        expect(utils.getCookieValue(expectedCookieName, document.cookie)).to.be.empty;
-        (new CallToAction($elm)).dismiss();
-        expect(utils.getCookieValue(expectedCookieName, document.cookie)).to.equal('true');
-      });
+            $fixture.parentElement.removeChild($fixture);
+          });
 
-      context('when the component\'s HTML element is missing an id attribute', () => {
+          it('is empty if the HTML element lacks a data-cookie-name-root attribute', function () {
+            const id = fixtures.generateRandomId();
+            const $fixture = fixtures.generateHTMLWithCookieDetails(fixtures.getCookieNameRoot(), id);
+            delete $fixture.dataset.cookieNameRoot;
 
-        let $elmNoId;
-        let callToActionWithElmNoId;
+            const callToAction = new CallToAction($fixture);
+            expect(callToAction.dismissible.cookieName).to.equal('');
 
-        beforeEach(() => {
-          $elmNoId = $elm.cloneNode(true);
-          $elmNoId.setAttribute('id', '');
-          $elm.parentElement.appendChild($elmNoId);
+            $fixture.parentElement.removeChild($fixture);
+          });
 
-          callToActionWithElmNoId = new CallToAction($elmNoId);
         });
 
-        afterEach(() => {
-          $elmNoId.parentElement.removeChild($elmNoId);
-        });
+        describe('expiry date', function () {
 
-        it('does not create a dismiss button', () => {
-          expect(callToActionWithElmNoId.$button).to.be.undefined;
-        });
+          it('is configured with the value of data-cookie-expires HTML attribute, if set', function () {
+            const id = fixtures.generateRandomId();
+            const expiry = 'Tue, 19 January 2038 03:14:07 UTC';
```

Comment:
this is just a string passed through so the code doesn't really care about the date format, but isn't a the format `Tue, 19 Jan 2038 03:14:07 UTC`? (looking at the `Expires` example in https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie)